### PR TITLE
Implement php7 features

### PR DESCRIPTION
**Pull request by NigelGreenway**
Adds

 - Scalar type hinting
 - Strict mode for all files
 - Return type hinting
 
Removes

 - Remove php 5.3 style code
 - Unused code

Changes

 - Corrects spelling mistakes

### CODE REVIEW

**Comment by NigelGreenway** [comment]
__Please don't merge until #3 has been merged so I am able to rebase once master is updated__

File: src/Server.php [hunk 1]
```diff
@@ -95,34 +93,34 @@ public function withHost($ip)
...
      *
-     * @param \Slim\App $app Slim application instance
+     * @param SlimInstance $app Slim application instance
      *
      * @return callable
      */
-    private function getCallbacks(\Slim\App $app)
+    private function getCallbacks(SlimInstance $app) :callable
     {
-        $server = $this;
```
Comment (NigelGreenway) [inline src/Server.php:104]: Please see [this](https://github.com/mbarquin/reactphp-slim/pull/3#issuecomment-297352067) in regards to how this should work going forward?

**Review by NigelGreenway — COMMENTED** [review]
As the package works from version 7+, this will work as intended.

**Comment by mbarquin** [comment]
Hi again,

The unused code found on the source is the beginning of the next release, I mean, I think this code is too much coupled, I would like to create adapters for the main objects (request, response, perhaps framework) on the project, and in future if there will be any other new server technology have the possiblity of writting different adapters for them.

**Comment by mbarquin** [comment]
Hi,

I have finished the tests and I have found two errors on phpDoc and declaration types,

* Request/SlimRequest.php:114 checkPartialUpload definition returns bool, but its a mixed return bool|string I think the return type should be removed.

* Request/SlimRequest.php:162 $name, $filename and $contentType are arrays not strings.

I think it's ok, after fixing this two issues I will go on with the merge :) :)

**Comment by NigelGreenway** [comment]
I will fix them parts today at some point.

Can you comment on the lines so I have a quick reference :)

**Comment by mbarquin** [comment]
[https://github.com/NigelGreenway/reactphp-slim/blob/feature/implement-php7-features/src/Request/SlimRequest.php#L162](https://github.com/NigelGreenway/reactphp-slim/blob/feature/implement-php7-features/src/Request/SlimRequest.php#L162)
Thnks, Change these definitons and after the merge I'll correct the comment :)

**Comment by NigelGreenway** [comment]
@mbarquin I've fixed the type hinting and have tested with a basic upload form.

Let me know if there are any issues with it. :)

**Comment by mbarquin** [comment]
Exccuse me I've been really busy, this night I will merge the code, and we can talk about your proposal for assets... thanks

**Comment by NigelGreenway** [comment]
👍

**Pull request merged**